language/move-vm/runtime/src/loader.rs

```diff
@@ -626,10 +626,10 @@ impl Loader {
     //
     // All modules in the bundle to be published must be loadable. This function performs all
     // verification steps to load these modules without actually loading them into the code cache.
-    pub(crate) fn verify_module_bundle_for_publication(
+    pub(crate) fn verify_module_bundle_for_publication<'a>(
         &self,
         modules: &[CompiledModule],
-        data_store: &mut impl DataStore,
+        data_store: &'a mut dyn DataStore,
```

Comment:
Why do we need to switch to `dyn`?

Reply:
I'm not sure about this. The following playground example compiles without any problems:
```
trait Foo {
    fn foo(&self);
}

fn g(x: &dyn Foo) {
    
}

fn f(x: &impl Foo) {
    g(x)
}
```
You can definitely use an `&imp`l as `&dyn`, but not the other way around due to the Sized constrant